src/QMCWaveFunctions/RotatedSPOs.h

```diff
@@ -340,11 +350,26 @@ class RotatedSPOs : public SPOSet, public OptimizableObject
   //  void evaluateThirdDeriv(const ParticleSet& P, int first, int last, GGGMatrix& grad_grad_grad_logdet)
   //  {Phi->evaluateThridDeriv(P, first, last, grad_grad_grad_logdet); }
 
+  /// Use history list (false) or global rotation (true)
+  void set_use_global_rotation(bool use_global_rotation) { use_global_rot_ = use_global_rotation; }
+
 private:
   /// true if SPO parameters (orbital rotation parameters) have been supplied by input
   bool params_supplied;
   /// list of supplied orbital rotation parameters
   std::vector<RealType> params;
+
+  /// Flag to ensure rotation is applied to shared coefficients only on one thread
+  bool use_this_copy_to_apply_rotation_;
```

Comment:
So if they are only applied on one thread what is the synchronization mechanism to ensure they aren't used on other threads before they are updated or during the update?  Assuming you'll be covered by some other rank scope synchronization then that needs to be documented.

I guess a future PR will make this clear?

Reply:
resetPsi explicitly iterates over all the clones - which is why no lock is needed, and why some solution like this flag is needed.  Another possible solution would be to add a flag to resetParameterExclusive to indicate the primary copy is being updated vs. one of the clones.

Reply:
I updated the PR to use an alternate method of passing a flag (isPrimaryObject) through resetParameterExclusive.

Reply:
@PDoakORNL and @ye-luo : Could I get your feedback on this latest commit (using the isPrimaryObject instead of use_this_copy... ).  I can put it in a separate PR for actual check-in.

Reply:
I prefer neither isPrimaryObject nor use_this_copy but to get rid of this line.
https://github.com/QMCPACK/qmcpack/blob/608db3e62129a0c3c7cdbc8db7be7ace282f747a/src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp#L444
Which needs fixing in a few optimizable components. All the optimization related operation should be done only through the gold copy. Updated parameters should be propagated automatically via shared_ptr among clones.

Reply:
I'd prefer this design direction as well.  I'm not fond of the shared pointer, from my perspective there isn't any ambiguity, the MultiBSplineSet or whatever basis is owned by the gold SPOSet. All the other SPOSet's should have lifetimes within the enclosing scope of the gold SPOSet lifetime. We don't need to do reference counting. The shared pointer is a quick workaround, fine for now but not ideal.

It doesn't really express the semantics and so doesn't indicate the undesirability of updating that pointed to object from what is basically a thread scoped object.